src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -27,21 +27,21 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             StorageCredentials credentials = account.Credentials;
 
             // Avoid double-validating the same account and credentials.
-            if (_validatedCredentials.Contains(credentials))
+            if ((!isPrimaryAccount && _validatedCredentials.Contains(credentials)) || _primaryCredentials == credentials)
             {
                 return;
             }
 
-            await ValidateCredentialsAsyncCore(account, cancellationToken);
+            await ValidateCredentialsAsyncCore(account, cancellationToken, isPrimaryAccount);
             _validatedCredentials.Add(credentials);
         }
 
-        private static async Task ValidateCredentialsAsyncCore(IStorageAccount account,
-            CancellationToken cancellationToken)
+        private async Task ValidateCredentialsAsyncCore(IStorageAccount account,
+            CancellationToken cancellationToken, bool isPrimaryAccount)
         {
             // Verify the credentials are correct.
             // Have to actually ping a storage operation.
-            IStorageBlobClient client = account.CreateBlobClient();
+            var client = account.CreateBlobClient();
```

Comment:
At the end, in C#, the compiled result will be the same, but when using `var` for a variable being set as a result of a method call, the type isn't clear, which would require others to inspect the return type (hover, etc.), but it's worse when looking at code outside of an IDE since you wouldn't have that option. Just makes things clear. If you're instantiating a type, though, `var` all day long :)